From 2020900bf1e47219e4c3e480b6cff8d4fa628e5d Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:23:18 -0400
Subject: [cpufreq] intel_pstate: Use LFM bus ratio as min ratio/P state

Message-id: <1409235879-1283-105-git-send-email-prarit@redhat.com>
Patchwork-id: 89414
O-Subject: [RHEL7.1 PATCH BZ 1134639 104/185] intel_pstate: Use LFM bus ratio as min ratio/P state
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 4042e7570cff740460b75c6fc604c629621d3dd2
Author: Dirk Brandewie <dirk.j.brandewie@intel.com>
Date:   Wed Feb 12 10:01:06 2014 -0800

    intel_pstate: Use LFM bus ratio as min ratio/P state

    LFM (max efficiency ratio) is the max frequency at minimum voltage
    supported by the processor.  Using LFM as the minimum P state
    increases performmance without affecting power. By not using P states
    below LFM we avoid using P states that are less power efficient.

    Signed-off-by: Dirk Brandewie <dirk.j.brandewie@intel.com>
    Cc: 3.13+ <stable@vger.kernel.org> # 3.13+
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/intel_pstate.c b/drivers/cpufreq/intel_pstate.c
index 1f4af61..61c26c7 100644
--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -357,7 +357,7 @@ static int byt_get_min_pstate(void)
 {
  u64 value;
  rdmsrl(BYT_RATIOS, value);
- return value & 0xFF;
+ return (value >> 8) & 0xFF;
 }
 
 static int byt_get_max_pstate(void)
-- 
1.7.1